File: 739-daily-temperatures/daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int n = temperatures.Length;
        Stack<int> stack = new Stack<int> ();
        int[] ans = new int[n] ;

        for (int r = 0; r < n ; r ++)
        {
            int temp = temperatures[r];

            while (stack.Count != 0 && temperatures[stack.Peek()] < temp)
            {
                int idx = stack.Pop();
                ans[idx] = r - idx;
            }

            stack.Push(r);


        }

        return ans;
        
    }
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        n = len(temperatures)
        ans = [0] * n


        for r in range(n):
            temp = temperatures[r]

            while stack and temperatures[stack[-1]]  < temp:
                idx = stack.pop() 
                ans[idx] = r - idx


            stack.append(r)
        
        return ans

        